eclair-core/src/main/scala/fr/acinq/eclair/io/Peer.scala

```diff
@@ -204,20 +181,25 @@ class Peer(val nodeParams: NodeParams, remoteNodeId: PublicKey, wallet: OnchainP
             stay()
         }
 
-      case Event(SpawnChannelNonInitiator(open, channelConfig, channelType, localParams), d: ConnectedData) =>
-        val channel = spawnChannel(None)
-        val temporaryChannelId = open.fold(_.temporaryChannelId, _.temporaryChannelId)
-        log.info(s"accepting a new channel with type=$channelType temporaryChannelId=$temporaryChannelId localParams=$localParams")
-        open match {
-          case Left(open) =>
-            channel ! INPUT_INIT_CHANNEL_NON_INITIATOR(open.temporaryChannelId, None, dualFunded = false, None, localParams, d.peerConnection, d.remoteInit, channelConfig, channelType)
-            channel ! open
-          case Right(open) =>
-            // NB: we don't add a contribution to the funding amount.
-            channel ! INPUT_INIT_CHANNEL_NON_INITIATOR(open.temporaryChannelId, None, dualFunded = true, None, localParams, d.peerConnection, d.remoteInit, channelConfig, channelType)
-            channel ! open
+      case Event(SpawnChannelNonInitiator(open, channelConfig, channelType, localParams, peerConnection), d: ConnectedData) =>
+        if (peerConnection == d.peerConnection) {
+          val channel = spawnChannel(None)
+          val temporaryChannelId = open.fold(_.temporaryChannelId, _.temporaryChannelId)
+          log.info(s"accepting a new channel with type=$channelType temporaryChannelId=$temporaryChannelId localParams=$localParams")
+          open match {
+            case Left(open) =>
+              channel ! INPUT_INIT_CHANNEL_NON_INITIATOR(open.temporaryChannelId, None, dualFunded = false, None, localParams, d.peerConnection, d.remoteInit, channelConfig, channelType)
+              channel ! open
+            case Right(open) =>
+              // NB: we don't add a contribution to the funding amount.
+              channel ! INPUT_INIT_CHANNEL_NON_INITIATOR(open.temporaryChannelId, None, dualFunded = true, None, localParams, d.peerConnection, d.remoteInit, channelConfig, channelType)
+              channel ! open
+          }
+          stay() using d.copy(channels = d.channels + (TemporaryChannelId(temporaryChannelId) -> channel))
+        } else {
+          // we got disconnected or reconnected and this message was for the previous connection
```

Comment:
I think the `ChannelClosed` event might not be the best choice because it requires parameters  that are not defined in `Peer`. If there were other actors watching for `ChannelClosed` they might do something unexpected if `Peer` emits a `ChannelEvent` produced with dummy values.

How about if `Peer` sends a message directly to `PendingChannelsRateLimiter` to remove a pending channel?

```scala
    log.warning(s"connection reset while waiting for open_channel to be accepted, ignoring")
    pendingChannelsRateLimiter ! RemoveChannelId(remoteNodeId, temporaryChannelId)
    stay()
```

Another option would be emit the new `ChannelAborted` message from PR #2593  .

Reply:
You're right, `ChannelClosed` isn't the right choice here. But `ChannelAborted` should be exactly what we want!

Reply:
Fixed in [816284b](https://github.com/ACINQ/eclair/pull/2552/commits/816284b4a03ee9772a85583810217f50bc4aff4b). Will not emit `ChannelAborted` when a peer reconnects before `OpenChannelInterceptor` accepts the channel.